aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -439,17 +437,34 @@ def parse_metrics_from_report_log(report_log_line):
     Returns:
         metrics - DatadogMetricPoint[]
     """
+    metrics = []
+    tags = ["cold_start:false"]
+
     regex_match = REPORT_LOG_REGEX.search(report_log_line)
 
     if not regex_match:
         return []
 
-    metrics = []
-    tags = []
+    # we always want coldstart and memorysize tags
+    tags.append("memorysize:" + regex_match.group(MEMORY_ALLOCATED_FIELD_NAME))
+
+    # if cold_start:
+    if regex_match.group(INIT_DURATION_METRIC_NAME):
```

Comment:
Nits, do you mind keeping the tags and metrics processing in separate sections, like the original code does? processing the tags first, then working on metrics?

I would leave the if..else... to generate the cold_start tag here, and move the logic to generate the init_duration metric to the existing for loop `For for metric_name in METRICS_TO_PARSE_FROM_REPORT` (probably you need to add a if statement inside the loop to handle the case when there is no regex match since init duration is not always present).